demos/text_detection_demo/src/text_recognition.cpp

```diff
@@ -25,6 +25,29 @@ namespace  {
         }
         *prob = 1.0f / static_cast<float>(sum);
     }
+
+    void softmax(const std::vector<float>::const_iterator& begin, const std::vector<float>::const_iterator& end, std::vector<float> &prob)
+    {
+        std::transform(begin, end, prob.begin(), static_cast<double(*)(double)>(std::exp));
+        float sum = std::accumulate(prob.begin(), prob.end(), 0.0f);
+        for (int i = 0; i < static_cast<int>(prob.size()); i++)
+            prob[i] /= sum;
+    }
+
+    struct BeamElement
+    {
+        std::string sentence;   //!< The sequence of chars that will be result of the beam element
```

Comment:
Could you use `vector<int>` instead of string here?
We considered with @Wovchena : It is natural to keep indexes in vectors of integers, not in strings.
(And IMHO integer type here is better than `char` type -- it would be required if we have strict memory restrictions)